tests/libvmi/lifecycle.go

```diff
@@ -44,3 +46,74 @@ func CleanupVMI(vmi *v1.VirtualMachineInstance) {
 		}, 2*time.Minute, time.Second).Should(SatisfyAll(HaveOccurred(), WithTransform(errors.IsNotFound, BeTrue())), "The VMI should be gone within the given timeout")
 	}
 }
+
+type VMISet map[string]*v1.VirtualMachineInstance
+
+func NewVMISet() VMISet {
+	return map[string]*v1.VirtualMachineInstance{}
+}
+
+func (vmis VMISet) Setup() {
```

Comment:
Working with the map in cases where you need only a single VMI would cause some unneeded boilerplate. e.g. I'd access the single VM using `vmis["vmi"]`. I'd prefer to keep the duplicity here and use the best fit for each test. Or do you see any way to get the best from both worlds?

Reply:
I was thinking something like:
```
func SetupVMI(vmi *v1.VirtualMachineInstance) *v1.VirtualMachineInstance {
	vmiSet := VMISet{"DummyVmiName": vmi}
	vmiSet.Setup()

	return vmiSet["DummyVmiName"]
}
```

Reply:
I see! Let me try it

Reply:
maybe "DummyVmiName" is too generic, since it's going to be on the `By` print.. then you could use vmi.name if possible

Reply:
Resolved, thanks a lot